src/memberPublicKey.spec.ts

```diff
@@ -0,0 +1,211 @@
+
+  let mockLogging: MockLogging;
+  let connection: Connection;
+  let serviceOptions: ServiceOptions;
+  let memberPublicKeyModel: ReturnModelType<typeof MemberPublicKeyModelSchema>;
+  beforeEach(() => {
+    mockLogging = makeMockLogging();
+    connection = getConnection();
+    serviceOptions = {
+      dbConnection: connection,
+      logger: mockLogging.logger,
+    };
+    memberPublicKeyModel = getModelForClass(MemberPublicKeyModelSchema, {
+      existingConnection: connection,
+    });
+  });
+
+  describe('createMemberPublicKey', () => {
+    test('Member public key should be created', async () => {
+      const memberPublicKey = await createMemberPublicKey(
+        MEMBER_MONGO_ID,
+        {
+          publicKey: publicKey.toString('base64'),
+          oid: TEST_OID,
+        },
+        serviceOptions,
+      );
+
+      requireSuccessfulResult(memberPublicKey);
+      const dbResult = await memberPublicKeyModel.findById(memberPublicKey.result.id);
+      expect(dbResult?.memberId).toStrictEqual(MEMBER_MONGO_ID);
+      expect(dbResult?.publicKey.toString('base64')).toStrictEqual(publicKey.toString('base64'));
+      expect(mockLogging.logs).toContainEqual(
+        partialPinoLog('info', 'Member public key created', {
+          id: memberPublicKey.result.id,
+        }),
+      );
+    });
+
+    test('Malformed public key should be refused', async () => {
+      const memberPublicKey = await createMemberPublicKey(
+        MEMBER_MONGO_ID,
+        {
+          publicKey: Buffer.from('invalid public key').toString('base64'),
+          oid: TEST_OID,
+        },
+        serviceOptions,
+      );
+
+      requireFailureResult(memberPublicKey);
+      expect(memberPublicKey.reason).toBe(MemberPublicKeyProblemType.MALFORMED_PUBLIC_KEY);
+    });
+
+    test('Record creation errors should be propagated', async () => {
+      await connection.close();
+
+      const error = await getPromiseRejection(
+        async () =>
+          createMemberPublicKey(
+            MEMBER_MONGO_ID,
+            {
+              publicKey: publicKey.toString('base64'),
+              oid: TEST_OID,
+            },
+            serviceOptions,
+          ),
+        Error,
+      );
+
+      expect(error).toHaveProperty('name', 'MongoNotConnectedError');
+    });
+  });
+
+  describe('getMemberPublicKey', () => {
+    test('Existing id should return the corresponding data', async () => {
+      const memberPublicKey = await memberPublicKeyModel.create({
+        memberId: MEMBER_MONGO_ID,
+        oid: TEST_OID,
+        publicKey,
+      });
+
+      const result = await getMemberPublicKey(
+        MEMBER_MONGO_ID,
+        memberPublicKey._id.toString(),
+        serviceOptions,
+      );
+
+      requireSuccessfulResult(result);
+      expect(result.result).toMatchObject({
+        oid: TEST_OID,
+        publicKey: publicKey.toString('base64'),
+      });
+    });
+
+    test('Non existent id should return non existing error', async () => {
```

Comment:
"Existent" seems to be grammatically correct, but most of our tests, including new tests in this PR, use "existing", so I think we should stick to it for consistency.